docs/dev_guide/version_control.rst

```diff
@@ -310,6 +318,59 @@ the resolution of the conflict with: ::
 
 You can then proceed to push this change up to your branch.
 
+Rebasing
+^^^^^^^^
+
+Sometimes it might be better to instead of merging in upstream/master, to rebase on top of upstream/master, or if you would like to clean up  your commit history if you deem it messy.
+**However**, be warned that rebasing is a nuclear option.
+If it goes wrong, it fundamentally changes your git history, there is no way back if you have not got a copy somewhere else, say your online fork of SunPy.
```

Comment:
You can just rebase on a different branch!
```bash
[MySuperFeature *]% git commit -m "My last messy commit"
[MySuperFeature]% git checkout -b MyCleanSuperFeature
[MyCleanSuperFeature]% git rebase -i upstream/master
```
and you still have your git history from `MySuperFeature` in its branch